src/execution_plan/ops/op_sort.c

```diff
@@ -161,6 +162,26 @@ static OpResult SortReset(OpBase *ctx) {
 	return OP_OK;
 }
 
+static OpBase *SortClone(const ExecutionPlan *plan, const OpBase *opBase) {
+	assert(opBase->type == OPType_SORT);
+	OpSort *op = (OpSort *)opBase;
+	OpSort *clone = rm_malloc(sizeof(OpSort));
+	clone->heap = NULL;
+	clone->buffer = NULL;
+	op->limit = op->limit;
+	array_clone(clone->directions, op->directions);
+
+	if(clone->limit) clone->heap = heap_new(_heap_elem_compare, clone);
```

Comment:
`if(clone->limit)` should probably be `if(op->limit)`

Reply:
I forgot to set `clone->limit`
fixed